### PR TITLE
dat.gui: Add default export. Add some JSDocs. Remove spurious addColor() overloads.

### PR DESCRIPTION
**Pull request by efokschaner**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - Docs for removal of missing addColor overload: https://github.com/dataarts/dat.gui/blob/master/API.md#GUI+addColor
  - Proof that implementation uses only `export default`: https://github.com/dataarts/dat.gui/blob/master/src/dat/index.js#L31
  - Docs for added JSDoc comments: https://github.com/dataarts/dat.gui/blob/master/API.md#new_GUI_new
- [x] Increase the version number in the header if appropriate. **Bumped to 0.7**
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/dat.gui/index.d.ts [hunk 1]
```diff
@@ -91,3 +117,12 @@ export class GUIController {
...
 }
+
+declare var index: {
+    gui: {
+        GUI: typeof GUI
+    },
+    GUI: typeof GUI
+};
+
+export default index;
```
Comment (weswigham) [inline types/dat.gui/index.d.ts]: > Proof that implementation uses only export default: https://github.com/dataarts/dat.gui/blob/master/src/dat/index.js#L31

[Their build system, however](https://github.com/dataarts/dat.gui/blob/master/build/dat.gui.js#L2519), seems to have other plans.

**Review by weswigham — CHANGES_REQUESTED** [review]
Their original source may use `default`, but their build chain transpiles it into a `module.exports =`; so you shouldn't state that this unconditionally use a default.

**Comment by efokschaner** [comment]
Hi @weswigham thanks for the review. So I think what's happening in my code is that webpack is importing from this other built file https://github.com/dataarts/dat.gui/blob/49d31a644ab26d9b1b71b101b769968f9c61c66b/build/dat.gui.module.js#L2513 which preserves the export default. In adding the export default I haven't removed the other non default exports which allows this typing to work for TS authors importing either built form as long as their import statement matches the style of the module they are bundling.

What are your thoughts on supporting both built forms?

**Comment by weswigham** [comment]
@efokschaner TS already let's a consumer state that their build system will do something like that using the `allowSyntheticDefaultImports` flag.

**Comment by efokschaner** [comment]
I remember trying that and not having luck, I'll double check it though. I think this might be the opposite problem I'm facing though. The compiler option says:
> Allow default imports from modules with no default export. This does not affect code emit, just typechecking.
Whereas in this situation the module (that I'm bundling) only has a default export whereas the typings do not say what it would be. Unless the compiler option means "no default export in their typings"...?

**Comment by weswigham** [comment]
You aughta be able to use a default import if the declaration file is written without one provided `allowSyntheticDefaultImports` is on. It does in fact mean "no default export in their typing". As it turns out, what's in the js has no bearing on the TS compilation if there's a .d.ts.

**Comment by efokschaner** [comment]
Right and it turns out `allowSyntheticDefaultImports` does work for me. Let me know what you think about whether having the typings work "out of the box" seeming preferable to having to enable this global compilation option. Given that dat.gui provides built modules compiled in both styles, having both export styles in the typings seems legitimate, and would make life easier for the next person in my situation.

If not, i'll pull that bit out. Thanks!

**Comment by weswigham** [comment]
You should pull that bit. It'll show up in completion lists for people not-using-webpack and given them runtime errors if used if you do not, which is generally to be avoided.

**Comment by efokschaner** [comment]
Ok, thanks @weswigham I've removed that bit.

**Pull request merged**